specs/agents/configuration.md

```diff
@@ -10,6 +10,31 @@ Here's a list of the config options across all agents, their types, default valu
 
 They are provided as environment variables but depending on the language there might be several feasible ways to let the user tweak them. For example besides the environment variable `ELASTIC_APM_SERVER_URL`, the Node.js Agent might also allow the user to configure the server URL via a config option named `serverUrl`, while the Python Agent might also allow the user to configure it via a config option named `server_url`.
 
+### Configuration Value Types
+
+The following table enumerates the available configuration types across the
+agents:
+
+
+| Type | Description (if needed) |
+|------|-------------------------|
+| String   |  |
+| Integer  |  |
+| Float    |  |
+| Boolean  | Encoded as a lower-case boolean string: `"false"`, `"true"` |
+| List     | Encoded as a comma-separated string: `"foo,bar,baz"` |
```

Comment:
Do we want to specify that whitespace around values is stripped? E.g. `"foo, bar\t , baz "` is interpreted as `["foo", "bar", "baz"]`?  The default value for `sanitize_field_names` in the Kibana central config UI (https://github.com/elastic/kibana/blob/master/x-pack/plugins/apm/common/agent_configuration/setting_definitions/general_settings.ts#L223-L228) is specified as comma-separate values *using spaces*, so I think APM agents need to do that whitespace trimming.